rst_files/lake_model.rst

```diff
@@ -669,84 +641,114 @@ Following :cite:`davis2006flow`, we set :math:`\alpha`, the hazard rate of leavi
 Fiscal Policy Code
 -----------------------
 
-We will make use of code we wrote in the :doc:`McCall model lecture <mccall_model>`, embedded below for convenience
+We will make use of code we wrote in the :doc:`McCall model lecture <mccall_model>`
+
+.. code-block:: julia
+
+    using Distributions, LinearAlgebra, Compat, Expectations, Parameters, NLsolve, Plots
+
+    function solve_mccall_model(mcm; U_iv = 1.0, V_iv = ones(length(mcm.w)), tol = 1e-5, 
+                                iter = 2_000)
+        # α, β, σ, c, γ, w = mcm.α, mcm.β, mcm.σ, mcm.c, mcm.γ, mcm.w
+        @unpack α, β, σ, c, γ, w, dist, u = mcm 
+        
+        # parameter validation
+        @assert c > 0.0
```

Comment:
Better to use `throw(ArgumentError(msg))`, but maybe best would be to use ArgCheck.jl

Reply:
No, the style guide says use `@assert` unless you are actually doing a proper error handling.  This is a test of an invariant.